libraries/react/src/basic-tests.js

```diff
@@ -198,6 +199,19 @@ describe("basic support", function () {
       expect(data).to.eql("React");
     });
 
+    it('will not overwrite methods', function () {
+      let root;
+      render(
+        <ComponentWithMethods
+          ref={(current) => {
+            root = current;
+          }}
+        />
+      )
+      let wc = root.wc;
+      expect(wc.innerText).to.eql('Success');
+    })
+
```

Comment:
Should probably mirror this to Preact too, but Preact will fail for a similar reason:

https://github.com/preactjs/preact/blob/5c9625a56b8d99e905b3beba9c09b5062fe41287/src/diff/props.js#L121-L124

This is more of a limitation in the standard JSX syntax than anything else, framework has to make a guess when it comes to attrs vs props.

Reply:
I'm happy to extend it to as many of them as I can manage; I've manged to get tests _mostly_ running with git bash on windows but it's still not fully happy.

Reply:
I might also reframe the test to catch other potential gotchas - readonly fields, etc.

Reply:
This is a great start, thanks @mrginglymus! 

@Westbrook Indeed, it would be great to get this implemented for all the frameworks.

Can we make a new branch for this, and then make this PR merge to that branch instead of `main`, that way we can get individual PRs merged piece by piece into the branch until all the tests for each framework, and then finally we can make a final PR to merge the branch into main once it is ready? Who has control of the repo to make that branch?

> This is more of a limitation in the standard JSX syntax than anything else, framework has to make a guess when it comes to attrs vs props.

@rschristian Solid.js and [Pota](https://pota.quack.uy/props/prop%3A__) both solve this within the limitation of JSX syntax, by looking for `attr:foo=`, `prop:foo=`, `bool:foo=`, and `on:foo=`. React and Preact *could* do something like too, without having to update the JSX spec.

Reply:
Oh! Here's an idea.

Perhaps, we don't have to get all tests for all frameworks implemented. We could consider any framework that's missing the test as not passing. Then this would encourage framework authors to chip in (or whoever to contribute it when they can).

Reply:
> React and Preact could do something like that too, without having to update the JSX spec, as `foo:bar=` is already valid JSX.

Barely.

`foo:` is a namespace, not supported by default in Babel (or any other tooling as far as I am aware) and not supported at all in many others. I write a PR to support it in Preact months ago but support is pretty limited for utilizing that.

Reply:
For what it's worth, `tsc` and `vite` seem to pass through namespaced attrs just fine. However, I don't think we need to solve React's problem here, just ensure that it's been recorded.

Reply:
Oh neat, good to know! I believe esbuild didn't support that in the past, unless I'm misremembering.